src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Represents a text JSON value.
+    /// </summary>
+    public class JsonString : JsonNode, IEquatable<JsonString>
+    {
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonString"/> class representing the empty value.
+        /// </summary>
+        public JsonString() => Value = "";
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonString"/> class representing a specified value.
+        /// </summary>
+        public JsonString(string value) => Value = value;
+
+        /// <summary>
+        ///   Gets or sets the text value represented by the instance.
+        /// </summary>
+        public string Value { get; set; }
+
+        /// <summary>
+        ///   Converts a <see cref="string"/> to a JSON string.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonString(string value) => new JsonString(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the value of this instance. 
+        /// </summary>
+        /// <param name="obj">The object to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the text value of this instance matches <paramref name="obj"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public override bool Equals(object obj) => obj is JsonString jsonString && Value == jsonString.Value;
```

Comment:
Same comment as in JsonBoolean, this should probably cast and then call the Equals method that you have bellow which also checks for null.